pandas/tests/indexes/test_base.py

```diff
@@ -305,6 +305,30 @@ def test_constructor_simple_new(self):
         result = idx._simple_new(idx, 'obj')
         tm.assert_index_equal(result, idx)
 
+    def test_constructor_names(self):
+        # test both `name` and `names` provided
+        with pytest.raises(TypeError):
+            idx = Index([1, 2, 3], name='a', names=('a',))
+
+        # test non-list-like `names`
+        with pytest.raises(TypeError):
+            idx = Index([1, 2, 3], names='a')
+
+        # test using `name` for a flat `Index`
+        idx = Index([1, 2, 3], name='a')
+        assert idx.name == 'a'
+        assert idx.names == ('a',)
+
+        # test using `names` for a flat `Index`
+        idx = Index([1, 2, 3], names=('a',))
+        assert idx.name == 'a'
+        assert idx.names == ('a',)
+
+        # test using `names` for a `MultiIndex`
+        idx = Index([('A', 1), ('A', 2)], names=('a', 'b'))
+        assert idx.name is None
```

Comment:
construct a MI and use ``tm.assert_index_equal``

Reply:
Do you mean I should construct MI explicitly? Or both cases (i.e. with `Index` and `MultiIndex` constructors)?